Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -8,33 +8,32 @@ enum ReportFormFocusField {
 
 @available(iOS 15.0, *)
 struct ReportProjectFormView: View {
+  @Binding var popToRoot: Bool
   let projectID: String
   let projectURL: String
   let projectFlaggingKind: GraphAPI.FlaggingKind
-  @Binding var popToRoot: Bool
 
   @SwiftUI.Environment(\.dismiss) private var dismiss
-  @ObservedObject private var viewModel = ReportProjectFormViewModel()
+  @StateObject private var viewModel = ReportProjectFormViewModel()
```

Comment:
@scottkicks OK, just learned something interesting. In a truly react-like way, `ReportProjectFormView` may be initialized many times - any time something calling it in a closure re-renders, for example. That means that `ReportProjectFormViewModel` was being initialized multiple times, which is a performance drag (and in the case of some re-renders, could also potentially mean we don't have the correct instance of the viewmodel). Apparently the solution is that the `View` should own the `ViewModel` using the `@StateObject` wrapper, which ensures that the view model is only instantiated once for the entire lifecycle of our view.

Reply:
This also means we can't pass in state directly on init of the viewmodel, or pass the view model directly into the view...so I set some properties on the viewmodel in `onAppear`.